packages/discovery/src/discovery/modelling/interpolate.ts

```diff
@@ -11,9 +11,10 @@ export function interpolateModelTemplate(
     tryCastingToName(String(values['$.address']), addressToNameMap, false),
   )
   const withValuesReplaced = withSelfReplaced.replace(
-    /#([a-zA-Z0-9_$.]+)(:raw)?/g,
-    (_match, key, raw) => {
+    /#([a-zA-Z0-9_$.]+)(:raw)?(\|lower)?/g,
```

Comment:
This is getting very close to being better to hand roll a parser for instead of tacking on additional regex stuff to it